imagely.es6

```diff
@@ -0,0 +1,242 @@
+import _ from 'lodash';
+import Promise from 'bluebird';
+import imageSize from 'image-size';
+import path from 'path';
+import phantom from 'phantom';
+import phantomjs from 'phantomjs';
+import request from 'request-promise';
+const fs = Promise.promisifyAll(require('fs'));
+
+/**
+ * Renders a source HTML file as an image.
+ *
+ * @public
+ * @param {String} source - HTML filepath or URL
+ * @param {String} destination - Destination image filepath; supported extensions are JPG, PNG, GIF, PDF
+ * @param {Object} [options]
+ * @param {Number} [options.width] - Viewport pixel width
+ * @param {Number} [options.height] - Viewport pixel height
+ * @param {Number} [options.scale=1] - Zoom level; use scale = 2 for HiDPI/Retina-ready output
+ * @param {String} [options.bg] - Background color
+ * @param {String} [options.json] - Filepath of JSON data to preload into window.data
+ * @param {Function} [callback] - Function to call upon completion; signature: (error, dimensions)
+ *                                where `dimensions` is an object with properties: { width, height }
+ */
+class Imagely {
+	constructor(source, destination, options, callback) {
+		Object.assign(this, { source, destination, options, callback });
+		this.imagely(source, destination, options, callback);
+	}
+
+	imagely(source, destination, options, callback) {
```

Comment:
I agree about keeping heavy lifting out of constructors and only doing `Object.assign(this, …parameters)`. However, that would mean we'd have to refactor all other methods to use `this.source`, `this.destination`, etc. instead of passing those parameters around—which looks like a followup at this point unless you'd prefer to tackle that now.